lib/go/edgecontext/validator.go

```diff
@@ -96,15 +119,37 @@ func (impl *Impl) validatorMiddleware(next secrets.SecretHandlerFunc) secrets.Se
 					err,
 				))
 			} else {
-				keys = append(keys, key)
+				if keys.first == nil {
+					keys.first = key
+				}
+				if fingerprint, err := RSAPublicKeyFingerprint(key); err != nil {
+					impl.logger.Log(context.Background(), fmt.Sprintf(
+						"Failed to get fingerprint of key #%d: %v",
+						i,
+						err,
+					))
+				} else {
+					keys.m[fingerprint] = key
+				}
 			}
 		}
 
-		if len(keys) == 0 {
+		if keys.first == nil {
 			impl.logger.Log(context.Background(), "No valid keys in secrets store.")
 			return
 		}
 
 		impl.keysValue.Store(keys)
 	}
 }
+
+// RSAPublicKeyFingerprint calculates the fingerprint of an RSA public key,
+// using ssh.FingerprintSHA256:
+// https://pkg.go.dev/golang.org/x/crypto/ssh#FingerprintSHA256
+func RSAPublicKeyFingerprint(pubKey *rsa.PublicKey) (string, error) {
```

Comment:
🔕 a lot of key-based functions take or return `interface{}` unfortunately -- you might want to check downstream usages of this to see if this will be the case there.  If it is, it would be nicer to accept `inerface{}` here and return an error if it's not the right type.

Reply:
This is only made public for the service issuing jwt tokens to use the same way of generating fingerprints. I think we do want explicit `*rsa.PublicKey` here because we already have hardcoded jwt algorithm of "RS256" and `jwt.ParseRSAPublicKeyFromPEM` calls so if we actually change to a different algorithm (and key format) that's a big breaking change anyways.